CMakeLists.txt

```diff
@@ -18,7 +18,13 @@ project(ruy CXX)
 cmake_minimum_required(VERSION 3.13)  # Copied from IREE
 set(CMAKE_CXX_STANDARD 14)
 
-option(RUY_ENABLE_TESTS "Enable ruy's tests" ON)
+if (PROJECT_NAME STREQUAL CMAKE_PROJECT_NAME)
```

Comment:
This appears to be the canonical way to do this: https://cliutils.gitlab.io/modern-cmake/chapters/testing.html

Maybe better?

Reply:
Thanks for bringing this up! I took a look, then I realized that I wanted something different for ruy:
1. The switch that I really want is not just "whether to build tests", it's "whether to build anything more than just the ruy public libraries", so it's not just about tests, it's also about the examples and in the future about docs.
2. What this page offers is not a drop-in replacement, because it results in having to replicate the `CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME` condition everywhere the `BUILD_TESTING` variable is checked. Indeed, in CTest.cmake, `BUILD_TESTING` is defined to have default value `ON` unconditionally. That in turns require another variable to override when people really want to enable tests in a dependency, so they end up suggesting that I write this:
```
if((CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME OR MYPROJECT_BUILD_TESTING) AND BUILD_TESTING)
    add_subdirectory(tests)
endif()
```
I prefer the way it's done in the commit that I just pushed, are you OK?  (i've tested it locally).

Reply:
I think not *building* things is mostly covered by `EXCLUDE_FROM_ALL` when included by a dependency. It won't get built unless someone specifically requests that target. The ifs are only necessary to avoid configuring things at all. Tests work a bit differently, for some reason in that they get added to the default ctest target even if they come from a subdirectory that is `EXCLUDE_FROM_ALL`, which is why they need the special handling. This seems fine though. I agree that having to check the project name everywhere would be a pain.